RadixWallet/Clients/URLFormatterClient/URLFormatterClient+Live.swift

```diff
@@ -23,10 +23,30 @@ extension URLFormatterClient: DependencyKey {
 	#endif
 
 	private static func makeURL(url: URL, imageServiceURL: URL, size: CGSize) -> URL {
-		let originItem = URLQueryItem(name: "imageOrigin", value: url.absoluteString)
-		let sizeItem = URLQueryItem(name: "imageSize", value: imageSizeString(size: size))
+		if url.isDataURL {
+			// Data URLs (e.g., `data:image/svg+xml,`) require special encoding to handle special characters.
+			// Here, we encode the absolute string using percent encoding, allowing only alphanumeric characters.
+			// This is needed because `url.appending(queryItems:)` does not encode special characters correctly for data URLs.
+			let imageOrigin = url.absoluteString.addingPercentEncoding(withAllowedCharacters: .alphanumerics) ?? ""
+			let imageSize = imageSizeString(size: size)
+
+			// Manually construct the query string to ensure that the data URL is correctly encoded
+			// and to avoid issues like double-encoding, which could corrupt the URL.
+			var urlString = "\(imageServiceURL.absoluteString)/?imageOrigin=\(imageOrigin)&imageSize=\(imageSize)"
+
+			if url.isVectorImage(type: .svg) {
+				urlString.append("&format=png")
+			}
+
+			return URL(string: urlString)!
```

Comment:
would it be better to default to original url (e.g. `return URL(string: urlString) ?? url)` instead of crashing, if for some reason we receive a weird url for which our manual construction fails?